pubsub/aws/snssqs/snssqs_test.go

```diff
@@ -0,0 +1,229 @@
+package snssqs
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_parseTopicArn(t *testing.T) {
+	// no further guarantees are made about this function
+	r := require.New(t)
+	r.Equal("qqnoob", parseTopicArn("arn:aws:sqs:us-east-1:000000000000:qqnoob"))
+}
+
+func Test_snsSqs_getSnsSqsMetatdata_happyPath(t *testing.T) {
```

Comment:
please name the test method specifically, avoid "happyPath". If too many constructs tested, please split the cases.